app/models/podcast.rb

```diff
@@ -14,10 +14,22 @@ class Podcast < BaseModel
 
   acts_as_paranoid
 
-  scope :published, -> { where('published_at IS NOT NULL') }
+  scope :published, -> { where('published_at IS NOT NULL AND published_at <= now()') }
 
-  after_save do
-    DateUpdater.last_build_date(self)
+  def pub_date
+    if max_episode_published_at
+      [published_at, max_episode_published_at].max
```

Comment:
Is there a reason to not just change the `Podcast.published_at` to be the max `Episode.published_at`, when the updated episode comes in?